ddtrace/contrib/internal/requests/connection.py

```diff
@@ -71,80 +68,83 @@ def _extract_query_string(uri):
 
 def _wrap_send(func, instance, args, kwargs):
     """Trace the `Session.send` instance method"""
-    # TODO[manu]: we already offer a way to provide the Global Tracer
-    # and is ddtrace.tracer; it's used only inside our tests and can
-    # be easily changed by providing a TracingTestCase that sets common
-    # tracing functionalities.
-    tracer = getattr(instance, "datadog_tracer", ddtrace.tracer)
-
-    # skip if tracing is not enabled
-    if not tracer.enabled and not asm_config._apm_opt_out:
-        return func(*args, **kwargs)
-
     request = get_argument_value(args, kwargs, 0, "request")
     if not request or is_otlp_export(request):
         return func(*args, **kwargs)
 
+    # Check if tracing is enabled (but allow if APM is opted out for ASM)
+    import ddtrace
+    from ddtrace.internal.settings.asm import config as asm_config
+
+    if not ddtrace.tracer.enabled and not asm_config._apm_opt_out:
+        return func(*args, **kwargs)
+
     url = _sanitized_url(request.url)
     method = ""
     if request.method is not None:
         method = request.method.upper()
     hostname, path = _extract_hostname_and_path(url)
     host_without_port = hostname.split(":")[0] if hostname is not None else None
 
-    cfg: Dict[str, Any] = {}
+    parsed_uri = parse.urlparse(url)
+
+    # Get pin from instance to access integration config
     pin = Pin.get_from(instance)
-    if pin:
-        cfg = pin._config
 
+    # Use pin's config if available, otherwise fall back to global config
+    integration_config = pin._config if pin else config.requests
```

Comment:
We can leave this in for now but we should leave a note that we will remove Pin in a future PR